src/HealthChecks.EventStore.gRPC/EventStoreHealthCheck.cs

```diff
@@ -0,0 +1,33 @@
+using EventStore.Client;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace HealthChecks.EventStore.gRPC
+{
+    public class EventStoreHealthCheck : IHealthCheck
+    {
+        private readonly string _connectionString;
+
+        public EventStoreHealthCheck(string connectionString)
+        {
+            _connectionString = connectionString ?? throw new ArgumentNullException(nameof(connectionString));
+        }
+
+        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                await using var client = new EventStoreClient(EventStoreClientSettings.Create(_connectionString));

```

Comment:
Are you sure to create new client on each HC call? What is the preferred lifetime of this client in app?

Reply:
The preferred lifetime is Singleton, so you're probably on to something.
I'll run some tests and make sure it's done correctly

Reply:
I've changed it to only create the client once, in the constructor.

Attached Benchmark in case you're curious about the performance of this 😄 
![image](https://user-images.githubusercontent.com/21295394/197627390-99066a44-3c7f-4a68-95dc-bad678a53267.png)